Tiltfile

```diff
@@ -395,14 +529,31 @@ def prepare_all():
                 context = context,
             )
 
+    addons_arg = ""
+    addon_substitutions = {}
+    for name in settings.get("deploy_addons", []):
+        addon = addons.get(name)
+        if addon == None:
+            fail("Addon with name " + name + " not found")
+        if addon.get("kustomize_config", True):
+            addon_substitutions = addon.get("kustomize_substitutions", {})
+            context = addon.get("context")
+            addons_arg = addons_arg + "--addons {name}:{container_name}:{binary_name}:{context} ".format(
+                name = name,
+                container_name = addon.get("container_name"),
+                binary_name = addon.get("binary_name"),
+                context = context,
+            )
     tilt_settings_file_arg = "--tilt-settings-file " + tilt_file
 
-    cmd = "make -B tilt-prepare && ./hack/tools/bin/tilt-prepare {allow_k8s_arg}{tools_arg}{cert_manager_arg}{kustomize_build_arg}{providers_arg}{tilt_settings_file_arg}".format(
+    os.environ["SERVICE_NAMESPACE"] = addon_substitutions.get("SERVICE_NAMESPACE", "default")
```

Comment:
I'm not sure this could work in this form, because we are running tilt prepare only once and SERVICE_NAMESPACE can be different for each addon

Reply:
Updated this so the test extension uses TEST_EXTENSION_SERVICE_NAMESPACE. The tiltfile also now creates one dictionary with all the variables from kustomize substitutions + addons and fails if there's a duplicate.